docs/source/library-user-guide/upgrading.md

```diff
@@ -490,6 +490,24 @@ If you were using a custom `SchemaAdapterFactory` for schema adaptation (e.g., d
 
 See the [default column values example](https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/custom_data_source/default_column_values.rs) for how to implement a custom `PhysicalExprAdapterFactory`.
 
+### `SchemaAdapter` and `SchemaAdapterFactory` completely removed
+
+The following symbols have been completely removed from DataFusion:
+
+- `SchemaAdapter` trait
+- `SchemaAdapterFactory` trait
+- `SchemaMapper` trait
+- `SchemaMapping` struct
+- `DefaultSchemaAdapterFactory` struct
+
+These types were previously used to adapt record batch schemas during file reading.
```

Comment:
This is likely to cause non trivial pain for anyone who uses the SchemaAdapter during upgrade

However, I am not sure if leaving the code in but disconnected would be any better.

Thus I think we should go with this PR and we can help with some more writeups when we start testing the upgrade with downstream crates (like delta.rs)

Reply:
How about I leave them in, mark them as deprecated and have them raise a runtime error with a link to the upgrading guide? At least then it doesn't fail silently at runtime. You'd have to ignore the compile time warnings as well.